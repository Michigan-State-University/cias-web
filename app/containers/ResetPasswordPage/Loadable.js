/**
 *
 * Asynchronously loads the component for ResetPasswordPage
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
