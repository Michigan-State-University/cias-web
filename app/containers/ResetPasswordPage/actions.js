import { actionBuilder } from 'utils/actionBuilder';
import {
  RESET_PASSWORD_REQUEST,
  RESET_PASSWORD_SUCCESS,
  RESET_PASSWORD_ERROR,
} from './constants';

export const resetPasswordRequest = (email) =>
  actionBuilder(RESET_PASSWORD_REQUEST, { email });
export const resetPasswordSuccess = () =>
  actionBuilder(RESET_PASSWORD_SUCCESS, {});
export const resetPasswordError = (error) =>
  actionBuilder(RESET_PASSWORD_ERROR, { error });
