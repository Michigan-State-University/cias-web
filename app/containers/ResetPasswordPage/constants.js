export const RESET_PASSWORD_REQUEST = 'app/auth/RESET_PASSWORD_REQUEST';
export const RESET_PASSWORD_SUCCESS = 'app/auth/RESET_PASSWORD_SUCCESS';
export const RESET_PASSWORD_ERROR = 'app/auth/RESET_PASSWORD_ERROR';
