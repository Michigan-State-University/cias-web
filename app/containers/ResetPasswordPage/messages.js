/*
 * ResetPasswordPage Messages
 *
 * This contains all the text for the ResetPasswordPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.ResetPasswordPage';

export default defineMessages({
  pageTitle: {
    id: `${scope}.pageTitle`,
    defaultMessage: 'Reset password',
  },
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Reset password',
  },
  validEmail: {
    id: `${scope}.validEmail`,
    defaultMessage: 'Provide valid email',
  },
  emailRequired: {
    id: `${scope}.emailRequired`,
    defaultMessage: 'Email is required',
  },
  emailPlaceholder: {
    id: `${scope}.emailPlaceholder`,
    defaultMessage: 'Enter e-mail address',
  },
  emailLabel: {
    id: `${scope}.emailLabel`,
    defaultMessage: 'E-mail address',
  },
  resetPassword: {
    id: `${scope}.resetPassword`,
    defaultMessage: 'Send a link',
  },
  return: {
    id: `${scope}.return`,
    defaultMessage: 'Back to login',
  },
  linkSent: {
    id: `${scope}.linkSent`,
    defaultMessage: 'Reset link was sent! Check your e-mail address.',
  },
});
