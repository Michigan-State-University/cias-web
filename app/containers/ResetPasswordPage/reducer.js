/*
 *
 * auth reducer
 *
 */
import produce from 'immer';

import {
  RESET_PASSWORD_REQUEST,
  RESET_PASSWORD_SUCCESS,
  RESET_PASSWORD_ERROR,
} from './constants';

export const initialState = {
  resetPasswordError: null,
  resetPasswordLoading: false,
};

/* eslint-disable default-case, no-param-reassign */
export const resetPasswordReducer = (state = initialState, { type, payload }) =>
  produce(state, (draft) => {
    switch (type) {
      case RESET_PASSWORD_REQUEST:
        draft.resetPasswordLoading = true;
        draft.resetPasswordError = null;
        break;
      case RESET_PASSWORD_SUCCESS:
        draft.resetPasswordLoading = false;
        draft.resetPasswordError = null;
        break;
      case RESET_PASSWORD_ERROR:
        draft.resetPasswordLoading = false;
        draft.resetPasswordError = payload.error;
        break;
    }
  });
