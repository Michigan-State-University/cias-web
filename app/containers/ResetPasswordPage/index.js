/**
 *
 * ResetPasswordPage
 *
 */

import React, { Fragment } from 'react';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import { compose } from 'redux';
import { Helmet } from 'react-helmet';
import { Formik } from 'formik';
import * as Yup from 'yup';

import { Fill } from 'components/Fill';
import Column from 'components/Column';
import H1 from 'components/H1';
import withPublicLayout from 'containers/PublicLayout';
import FormikInput from 'components/FormikInput';
import Button from 'components/Button';
import LinkButton from 'components/Button/LinkButton';

import messages from './messages';

const validationSchema = formatMessage =>
  Yup.object().shape({
    email: Yup.string()
      .required(formatMessage(messages.emailRequired))
      .email(formatMessage(messages.validEmail)),
  });

const initialValues = {
  email: '',
};

const ResetPasswordPage = ({ intl: { formatMessage } }) => {
  const onSubmit = () => {};
  return (
    <Fragment>
      <Helmet>
        <title>{formatMessage(messages.pageTitle)}</title>
      </Helmet>
      <Fill justify="center" align="center">
        <Column sm={10} md={8} lg={6} align="start">
          <H1 mb={40} fontSize={23}>
            <FormattedMessage {...messages.header} />
          </H1>
          <Formik
            validationSchema={validationSchema(formatMessage)}
            initialValues={initialValues}
            onSubmit={onSubmit}
          >
            {({ handleSubmit }) => (
              <Fragment>
                <FormikInput
                  formikKey="email"
                  placeholder={formatMessage(messages.emailPlaceholder)}
                  type="text"
                  label={formatMessage(messages.emailLabel)}
                  inputProps={{
                    height: 46,
                    width: '100%',
                  }}
                />
                <Button
                  height={46}
                  borderRadius={5}
                  mt={45}
                  loading={false}
                  onClick={handleSubmit}
                  type="submit"
                >
                  <FormattedMessage {...messages.resetPassword} />
                </Button>
                <LinkButton to="/login" mt={25} alignSelf="center">
                  <FormattedMessage {...messages.return} />
                </LinkButton>
              </Fragment>
            )}
          </Formik>
        </Column>
      </Fill>
    </Fragment>
  );
};

ResetPasswordPage.propTypes = {
  intl: intlShape,
};

const mapDispatchToProps = {};

const withConnect = connect(
  null,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  injectIntl,
  withPublicLayout,
)(ResetPasswordPage);
