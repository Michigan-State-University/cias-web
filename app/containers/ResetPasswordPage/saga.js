import axios from 'axios';
import { put, takeLatest, call } from 'redux-saga/effects';
import { toast } from 'react-toastify';

import { formatMessage } from 'utils/intlOutsideReact';

import { resetPasswordError, resetPasswordSuccess } from './actions';
import { RESET_PASSWORD_REQUEST, RESET_PASSWORD_SUCCESS } from './constants';
import messages from './messages';

function* resetPassword({ payload: { email } }) {
  const requestURL = `/v1/auth/password`;
  try {
    yield axios.post(requestURL, {
      email,
      redirect_url: `${process.env.WEB_URL}/set-new-password`,
    });
    yield call(toast.success, formatMessage(messages.linkSent), {
      toastId: RESET_PASSWORD_SUCCESS,
    });
    yield put(resetPasswordSuccess());
  } catch (error) {
    yield put(resetPasswordError(error?.toString()));
  }
}

export default function* resetPasswordSaga() {
  yield takeLatest(RESET_PASSWORD_REQUEST, resetPassword);
}
