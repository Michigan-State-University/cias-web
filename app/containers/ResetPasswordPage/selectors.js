import { createSelector } from 'reselect';

import { initialState } from './reducer';

const selectAuth = state => state.resetPassword || initialState;

export const makeSelectError = () =>
  createSelector(
    selectAuth,
    substate => substate.resetPasswordError,
  );

export const makeSelectLoader = () =>
  createSelector(
    selectAuth,
    substate => substate.resetPasswordLoading,
  );
