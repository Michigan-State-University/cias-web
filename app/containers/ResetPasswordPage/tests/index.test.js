/**
 *
 * Tests for ResetPasswordPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { MemoryRouter } from 'react-router-dom';

import ResetPasswordPage from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<ResetPasswordPage />', () => {
  const reducer = state => state;
  const initialState = {};
  let store;
  beforeAll(() => {
    store = createStore(reducer, initialState);
    store.runSaga = () => {};
    store.injectedReducers = {};
    store.injectedSagas = {};
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <ResetPasswordPage />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter>
            <ResetPasswordPage />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
