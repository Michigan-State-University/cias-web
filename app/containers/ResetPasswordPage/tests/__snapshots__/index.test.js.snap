// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordPage /> Should render and match the snapshot 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c7 {
  margin: 0;
  font-family: DM Sans;
  font-size: 25px;
  line-height: 130%;
  margin-bottom: 40px;
  font-size: 23px;
  color: #2F3850;
}

.c3 {
  width: auto;
  height: auto;
  height: 80px;
}

.c0 {
  position: relative;
  z-index: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 100%;
  min-height: 100%;
  background: #F8FBFF;
}

.c1 {
  position: absolute;
  z-index: -1;
  background: #FFFFFF;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0.75;
  -ms-flex: 0.75;
  flex: 0.75;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  padding: 0 20px 30px 20px;
  border-radius: 5px;
  box-shadow: 0px 4px 10px rgba(227,238,251,0.8);
  position: fixed;
  height: 100vh;
  width: 50vw;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 30px 15px;
  margin-left: 50vw;
}

.c11 {
  padding: 12px;
  border-style: solid;
  border-width: 1px;
  border-color: #E9F1FB;
  border-radius: 5px;
  width: 100%;
  height: 46px;
  color: #2F3850;
}

.c11:focus {
  box-shadow: none;
  outline: none;
  border-color: #107969;
}

.c11:disabled {
  color: #ABBBD1;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c9 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 5px;
  font-size: 12px;
  color: #2F3850;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c12 {
  width: 100%;
  height: 40px;
  border-radius: 100px;
  cursor: pointer;
  border: none;
  outline: none;
  font-weight: bold;
  background-color: #107969;
  color: #FFFFFF;
  border: 1px solid transparent;
  -webkit-transition: background-color 300ms ease,color 300ms ease,border 300ms ease;
  transition: background-color 300ms ease,color 300ms ease,border 300ms ease;
  margin-top: 45px;
  height: 46px;
}

.c14 {
  outline: none;
  border: none;
  background: transparent;
  font-weight: bold;
  font-size: 13px;
  line-height: 17px;
  color: #B727EA;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c13 {
  margin-top: 25px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media only screen and (min-width:768px) {
  .c6 {
    width: 83.33333333333334%;
  }
}

@media only screen and (min-width:960px) {
  .c6 {
    width: 66.66666666666666%;
  }
}

@media only screen and (min-width:1170px) {
  .c6 {
    width: 50%;
  }
}

@media only screen and (min-width:1360px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

@media only screen and (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    background: transparent;
    height: 400px;
    width: 100%;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    background: #FFFFFF;
    position: relative;
    height: auto;
    width: 100%;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    background: none;
    padding: 30px 15px;
    -webkit-flex: initial;
    -ms-flex: initial;
    flex: initial;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-left: 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <img
        alt="CIAS Application Logo"
        class="c3"
        height="80"
        src="IMAGE_MOCK"
      />
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <h1
            class="c7"
            font-size="23"
          >
            Reset password
          </h1>
          <div
            class="c8"
          >
            <label
              for="email"
            >
              <p
                class="c9"
                font-size="12px"
                width="fit-content"
              >
                E-mail address
              </p>
            </label>
            <div
              class="c10"
              width="100%"
            >
              <input
                class="c11"
                height="46"
                id="email"
                name="email"
                placeholder="Enter e-mail address"
                type="text"
                value=""
                width="100%"
              />
            </div>
          </div>
          <button
            class="c12"
            color="primary"
            height="46"
            radius="100px"
            type="submit"
          >
            Send a link
          </button>
          <div
            class="c13"
          >
            <a
              class="c14"
              href="/login"
            >
              Back to login
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
